crates/uv/src/commands/publish.rs

```diff
@@ -0,0 +1,74 @@
+use crate::commands::{human_readable_bytes, ExitStatus};
+use crate::printer::Printer;
+use anyhow::{bail, Result};
+use owo_colors::OwoColorize;
+use std::fmt::Write;
+use tracing::info;
+use url::Url;
+use uv_client::{BaseClientBuilder, Connectivity};
+use uv_configuration::{KeyringProviderType, TrustedHost};
+use uv_publish::{files_for_publishing, upload};
+
+pub(crate) async fn publish(
+    paths: Vec<String>,
+    publish_url: Url,
+    keyring_provider: KeyringProviderType,
+    allow_insecure_host: Vec<TrustedHost>,
+    username: Option<String>,
+    password: Option<String>,
+    connectivity: Connectivity,
+    native_tls: bool,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if connectivity.is_offline() {
+        bail!("You cannot publish files in offline mode");
+    }
+
+    let files = files_for_publishing(paths)?;
+    match files.len() {
+        0 => bail!("No files found to publish"),
+        1 => writeln!(printer.stderr(), "Publishing 1 file")?,
+        n => writeln!(printer.stderr(), "Publishing {n} files")?,
+    }
+
+    let client = BaseClientBuilder::new()
+        // https://github.com/seanmonstar/reqwest/issues/2416
+        .retries(0)
+        .keyring(keyring_provider)
+        .native_tls(native_tls)
+        .allow_insecure_host(allow_insecure_host)
+        // Don't try cloning the request to make an unauthenticated request first.
+        // https://github.com/seanmonstar/reqwest/issues/2416
```

Comment:
Seems like we could comment on this once above instead of linking to the issue twice?